gapii/cc/vulkan_extras.cpp

```diff
@@ -995,4 +1003,28 @@ void VulkanSpy::walkImageSubRng(
     }
   }
 }
+
+void VulkanSpy::recordWaitedSemaphores(CallObserver* observer, VkDevice device,
+                                       const VkSemaphoreWaitInfo* wait_info,
+                                       bool isKHR) {
```

Comment:
I don't think we actually need to save the semaphore values here. I think replay already has all the information we need, even if only a subset of the semaphores are signaled (either because flags contains `VK_SEMAPHORE_WAIT_ANY_BIT`, or because `vkWaitSemaphores` returned `VK_TIMEOUT`).

At replay, we know the semaphore values after all the unblocked commands, so, if we wanted to be conservative, we could replay a `WAIT_ANY` as a `WAIT_ALL`, where the semaphores we know are unsignaled are filtered out.